fhir-r-lang/exercise_1.Rmd

```diff
@@ -0,0 +1,463 @@
+
+In this use case we're interested in patients who've been prescribed opioids. Looking at the [FHIR Patient profile](https://www.hl7.org/fhir/patient.html) we see a few references to things like their primary care provider or the organization managing the patient record, but nothing about medications. Luckily, above this profile we see a list of other resources that reference Patient - including [MedicationRequest](https://www.hl7.org/fhir/medicationrequest.html#MedicationRequest). Looking within MedicationRequest we see that `MedicationRequest.subject` identifies the patient the medication is for.
+
+Let's see if we can find the Medications prescribed to the patient with id `10098`. The core FHIR spec doesn't require that any specific searches be supported so it's important to read the documentation and look at the CapabilityStatement of the server being queried to get an idea of the options available. For now we'll look at the core FHIR spec for an idea of search parameters defined in the base spec that servers might implement.
+
+Looking at the MedicationRequest Resource core FHIR documentation it looks like there are two search parameters that would be helpful: `patient` and `subject`. Practically either would work just fine, but looking at the Expression we can see that `patient` only works for references to a Patient resource, while `subject` would work for references to either a Patient or a Group. Looking at the CapabilityStatement, it also appears that the server supports both!
+
+_Note: Specific FHIR Implementation Guides, like US Core, may define their own SearchParameters for servers to implement_
+
+📘[Read more about FHIR Resource Organization](https://www.hl7.org/fhir/overview-arch.html#organizing)
+
+```{r}
+(cs$Resources %>% filter(type == "MedicationRequest"))$searchParam.name %>%
+  str_replace_all(" \\|\\| ", "\n") %>% cat
+```
+
+Let's pull in the [MedicationRequest](https://hl7.org/fhir/medicationrequest.html) instances for one patient using the JSON approach -- this gives us an easy way to see which elements are provided by this server. You can also look at the [MedicationRequest spec](https://hl7.org/fhir/medicationrequest.html) to help determine which elements are populated.
+
+```{r}
+response <- httr::GET(
+  url = str_interp("${fhir_server}/MedicationRequest?patient=10098"),
+  config = list(add_headers( Accept = 'application/fhir+json'))
+)
+
+# Convert from raw `httr` response into an R list for easier access
+response_list <- fromJSON(httr::content(response, as = "text", encoding = "UTF-8"), flatten = TRUE)
+
+response_list$entry %>% glimpse
+```
+
+Now we can use `fhircrackr` to quickly load these into a data frame:
+
+```{r}
+request <- fhir_url(url = fhir_server, resource = "MedicationRequest", parameters = c(patient="10098"))
+medication_request_bundle <- fhir_search(request = request)
+
+# Identify which elements of the FHIR resource we want to capture in our data frame - see Exercise 0 for details
```

Comment:
Can we print the JSON here to give attendees and idea of what the data looks like before putting it in a dataframe?

Reply:
Done